docs/developers/testing.md

```diff
@@ -0,0 +1,314 @@
+```
+python run_core_tests.py \
+    --project-directory projects/OS-OS_VERSION/OS-OS_VERSION-DATABASE-DATABASE_VERSION \
+    --irods-package-directory /path/to/directory/containing/irods/package/files
+```
+
+#### Using concurrency to reduce test time
+
+The `--concurrent-test-executors` option allows the user to specify how many iRODS zones to deploy and use for splitting up the list of tests to run. Note that this option can quickly increase the number of containers created by the Compose project (2 containers per test executor), so make sure to choose a number appropriate to the host system's capabilities. Here is an example of how to use the option:
+```
+python run_core_tests.py \
+    --project-directory projects/OS-OS_VERSION/OS-OS_VERSION-DATABASE-DATABASE_VERSION \
+    --concurrent-test-executors 4
+```
+This will deploy 4 iRODS zones (4 * (1 catalog service provider + 1 database) = 8 containers) and will run tests from the list of tests by taking them from a queue and concurrently executing on the catalog service providers in each zone.
+
+The test suite takes 8-10 hours to complete when run serially. With 4 concurrent executors on an 8-core host system, the test suite takes about 2 hours to complete.
+
+#### Running specific tests
+
+The `--tests` option allows the user to specify a space-delimited list of specific tests to run. Here is an example of how to use the option:
+```
+python run_core_tests.py \
+    --project-directory projects/OS-OS_VERSION/OS-OS_VERSION-DATABASE-DATABASE_VERSION \
+    --tests test_ils test_icp test_ihelp
+```
+This will deploy a single iRODS zone and execute the following tests in order on the catalog service provider: `test_ils`, `test_icp`, and `test_ihelp`. This can be used in conjunction with the `--concurrent-test-executors` option, but the tests will not be run in a deterministic order as the concurrent executors will take tests from the queue as they complete them.
+
+### How to run unit tests: `run_unit_tests.py`
+
+`run_unit_tests.py` is the main script for running the iRODS "unit" test suite (see [Unit test suite](#unit-test-suite-catch2)).
+
+Here is the minimal form for running the test suite for a particular `OS`, `OS_VERSION`, `DATABASE`, and `DATABASE_VERSION`:
+```
+python run_unit_tests.py --project-directory projects/OS-OS_VERSION/OS-OS_VERSION-DATABASE-DATABASE_VERSION
+```
+
+This will serially run the unit test suite using the latest release of the iRODS server. This will create a container for the iRODS server configured as the sole catalog service provider, a container for the database where the catalog is stored, and a network so that they can communicate as if it were a distributed environment. A running iRODS server is required because the Catch2 tests are not actually unit tests. Most of the same options available for `run_core_tests.py` are available for this script as well.
```

Comment:
```suggestion
This will serially run the unit test suite using the latest release of the iRODS server. This will create a container for the iRODS server configured as the sole catalog service provider, a container for the database where the catalog is stored, and a network so that they can communicate as if it were a distributed environment. A running iRODS server is required because the Catch2 tests are not actually unit tests. Most of the same options available for `run_core_tests.py` are also available for this script.
```